src/components/SettingsDialog/Participants/__tests__/Participants.test.tsx

```diff
@@ -16,4 +16,18 @@ describe("Participants", () => {
     const {container} = render(createParticipants(getTestStore()));
     expect(container.firstChild).toMatchSnapshot();
   });
+
+  describe("Ban Feature", () => {
+    test("ban button not available on self", () => {
+      const {container} = render(createParticipants(getTestStore()));
+      const participants = container.getElementsByClassName("participants__list")[0].childNodes;
+      expect(participants[0].lastChild).not.toHaveClass("participant__kick-icon");
```

Comment:
In my opinion, it's better to search for elements from the end user's point of view, e.g. using labels, placeholders, etc.

So in this case, you could do something like this:
```
import {within} from "@testing-library/react";

expect(within(participants[0].lastChild).findByLabelText(t("Participants.BanParticipantTooltip"))).not.toBeInTheDocument();
});
```

Reply:
I implemented it now as follows:
```tsx
const participants = container.getElementsByClassName("participants__list-item");
expect(within(participants[1]).getByLabelText(t("Participants.BanParticipantTooltip"))).toBeInTheDocument();
```

However, TypeScript throws a warning as `participants` is of type `HTMLElementsCollectionOf<Element>` but the `within` function requires an `HTMLElement`.
The test completes successfully, but if you have a better way be my guest.